frontend/lib/src/components/elements/ArrowVegaLiteChart/arrowUtils.ts

```diff
@@ -164,25 +162,26 @@ export function getDataArray(
         typeof indexValue === "bigint" ? Number(indexValue) : indexValue
     }
 
-    for (let colIndex = 0; colIndex < cols; colIndex++) {
+    for (let colIndex = 0; colIndex < numColumns; colIndex++) {
       const dataValue = quiverData.getDataValue(rowIndex, colIndex)
       const dataType = quiverData.columnTypes.data[colIndex]
-      const typeName = getTypeName(dataType)
 
       if (
-        typeName !== "datetimetz" &&
         (dataValue instanceof Date || Number.isFinite(dataValue)) &&
-        (typeName.startsWith("datetime") || typeName === "date")
+        (isDatetimeType(dataType) || isDateType(dataType)) &&
+        // Only convert dates without timezone information
+        // to utc timezone
+        !getTimezone(dataType)
       ) {
         // For dates that do not contain timezone information.
         // Vega JS assumes dates in the local timezone, so we need to convert
         // UTC date to be the same date in the local timezone.
         const offset = new Date(dataValue).getTimezoneOffset() * 60 * 1000 // minutes to milliseconds
         row[quiverData.columnNames[0][colIndex]] = dataValue.valueOf() + offset
-      } else if (typeof dataValue === "bigint") {
-        row[quiverData.columnNames[0][colIndex]] = Number(dataValue)
       } else {
-        row[quiverData.columnNames[0][colIndex]] = dataValue
+        // VegaLite can't handle BigInts, so they have to be converted to Numbers first
```

Comment:
To understand: we transfer all bigints to `Number` because `VegaLite` does not support it, right? Because I don't see any check whether this is for VegaLite.
Additionally, it might make sense to state here or somewhere more prominent that we might lose accuracy or get entirely wrong results.

Reply:
The arrowUtils here are Vega lite specific (see the path)

Reply:
> The arrowUtils here are Vega lite specific (see the path)

oh yeah, I missed that!